v3/lint/base.go

```diff
@@ -200,9 +205,7 @@ func (l *CertificateLint) toLint() *Lint {
 // if IneffectiveDate is zero then only EffectiveDate is checked. If both EffectiveDate
 // and IneffectiveDate are zero then CheckEffective always returns true.
 func (l *CertificateLint) CheckEffective(c *x509.Certificate) bool {
-	onOrAfterEffective := l.EffectiveDate.IsZero() || util.OnOrAfter(c.NotBefore, l.EffectiveDate)
-	strictlyBeforeIneffective := l.IneffectiveDate.IsZero() || c.NotBefore.Before(l.IneffectiveDate)
-	return onOrAfterEffective && strictlyBeforeIneffective
+	return checkEffective(l.EffectiveDate, l.IneffectiveDate, c.NotBefore)
```

Comment:
This datetime range checking function has had a history of causing trouble due to off-by-one errors caused by general confusion on the semantics of the function.

So, in order to reduce the surface area caused by having the logic copied in two spots, I moved it's body out to a private utility function located at the end of this file.